pkg/k8s/client/cell.go

```diff
@@ -545,6 +587,86 @@ func FakeClientBuilder() ClientBuilderFunc {
 	}
 }
 
+func FakeClientCommand(fc *FakeClientset) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "interact with fake k8s client",
+			Args:    "<command> args...",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			if len(args) < 1 {
+				return nil, fmt.Errorf("usage: k8s <command> files...\n<command> is one of add, update or delete.")
+			}
+
+			action := args[0]
+			if len(args) < 2 {
+				return nil, fmt.Errorf("usage: k8s %s files...", action)
+			}
+
+			for _, file := range args[1:] {
+				b, err := os.ReadFile(s.Path(file))
+				if err != nil {
+					// Try relative to current directory, e.g. to allow reading "testdata/foo.yaml"
+					b, err = os.ReadFile(file)
+				}
+				if err != nil {
+					return nil, fmt.Errorf("failed to read %s: %w", file, err)
+				}
+				obj, gvk, err := testutils.DecodeObjectGVK(b)
+				if err != nil {
+					return nil, fmt.Errorf("decode: %w", err)
+				}
+				gvr, _ := meta.UnsafeGuessKindToResource(*gvk)
+				objMeta, err := meta.Accessor(obj)
+				if err != nil {
+					return nil, fmt.Errorf("accessor: %w", err)
+				}
+				name := objMeta.GetName()
+				ns := objMeta.GetNamespace()
+
+				// Try to add the object to all the trackers. If one of them
+				// accepts we're good. We'll add to all since multiple trackers
+				// may accept (e.g. slim and kubernetes).
+
+				// err will get set to nil if any of the tracker methods succeed.
+				// start with a non-nil default error.
+				err = errors.New("fail")
+				checkErr := func(e error) {
+					if err != nil {
+						err = e
+					}
+				}
+				switch action {
```

Comment:
Nit: could we create a slice `[]testing.ObjectTracker` containing all of `fc.XXX.Tracker()`s before the switch and then just iterate over it for each case?

Reply:
Good idea! Changed it so that the trackers are collected where the fake clients are created.. that way it's easy to extend.